x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java

```diff
@@ -97,35 +112,44 @@ public void testRemove_GivenJobsWithMixedRetentionPolicies() {
         Date oneDayAgo = new Date(now.getTime() - TimeValue.timeValueDays(1).getMillis());
         SearchHit snapshot1_1 = createModelSnapshotQueryHit("job-1", "fresh-snapshot", oneDayAgo);
         searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshot1_1)));
+        SearchHit snapshot2_1 = createModelSnapshotQueryHit("job-2", "fresh-snapshot", oneDayAgo);
+        searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshot2_1)));
 
         // It needs to be strictly more than 7 days before the most recent snapshot, hence the extra millisecond
         Date eightDaysAndOneMsAgo = new Date(now.getTime() - TimeValue.timeValueDays(8).getMillis() - 1);
-        SearchHit snapshotToBeDeleted = createModelSnapshotQueryHit("job-1", "old-snapshot", eightDaysAndOneMsAgo);
-
-
-        searchResponses.add(
-            AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshotToBeDeleted)));
-
-        SearchHit snapshot2_1 = createModelSnapshotQueryHit("job-1", "snapshots-1_1", eightDaysAndOneMsAgo);
```

Comment:
I am not sure how this test ever passed?

I think it is meant to test the mixed retention between two jobs, but it appears to give the same job two of the snapshots?

I adjusted it it for the new format so that this other old snapshot is part of `job-2`

Reply:
@droberts195 I will update my mocks so that for `job-2` nothing is returned